compiler/next/lib/types/BuiltinType.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/types/BuiltinType.h"
+
+namespace chpl {
+namespace types {
+
+
+void BuiltinType::canonicalizeBitWidth() {
+  switch (kind()) {
+    case BOOL:
+    case INT:
+    case UINT:
+      if (bitwidth_ == 0) {
+        if (kind() == BOOL) {
```

Comment:
It seems a little unfortunate to group these three together and then immediately check about one of them, but I personally can't think of a different organization that isn't confusing.

Reply:
Yeah, this function went through a few iterations and I couldn't think of a better organization either.